src/plugins/intel_cpu/src/nodes/kernels/gather_uni_kernel.hpp

```diff
@@ -69,142 +70,233 @@ struct gatherJitExecArgs {
     uint64_t betweenBatchAndAxisIter;
 };
 
-struct jitGatherKernelBase {
-    void (*ker_)(const gatherJitExecArgs *);
-    void operator()(const gatherJitExecArgs *args) {
+struct jitGatherKernelInterface {
+    virtual ~jitGatherKernelInterface() = default;
+    virtual void initialize(const jGatherConfParams& jcp) = 0;
+    virtual bool isSameParams(const jGatherConfParams& jcp) = 0;
+    virtual void operator()(const gatherJitExecArgs *args) = 0;
+    virtual void create_ker() = 0;
+    virtual uint64_t getVecLen() const = 0;
+    virtual uint64_t getDataElPerVec() const = 0;
+    virtual uint64_t getIdxElPerVec() const = 0;
+};
+
+using namespace dnnl::impl::cpu;
+
+enum DataTypeSize {
+    DataType32bit,
+    DataType16bit,
+    DataType8bit
+};
+
+enum Approach {
+    Long,
+    Short
+};
+
+enum AfterAxisCase {
+    ElementwiseCase,
+    BlockedCase
+};
+
+enum NodeShapesDynamicity {
+    DynamicShapes,
+    StaticShapes
+};
+
+std::shared_ptr<jitGatherKernelInterface> createJitUniGatherKernel(x64::cpu_isa_t isa,
+    uint64_t dataTypeSize, bool isDynamicNode, uint64_t afterAxisSize, uint64_t specIdxSize, uint64_t idxElPerVec);
+
+template <x64::cpu_isa_t isa>
+class jitGatherKernelForIsa;
```

Comment:
Why duplicate Isa in name of class ?
As for me `jitGatherKernel<avx512_core>` is also readable

Reply:
It could be confusing because the jitGatherKernelForIsa is an abstract base class and is not supposed for instantiation. Would the `jitGatherKernelBase` be better?

Reply:
If it is not supposed for instantiation then `jitGatherKernelBase` is more readable as for me

Reply:
Ok, I changed it to `jitGatherKernelBase`.